### PR TITLE
Json runner

### PR DESCRIPTION
**Pull request by bhalonen**
It is still an a ugly beast, but give these changes a glance.

### CODE REVIEW

File: include/Strain2D.h [hunk 1]
```diff
@@ -105,8 +105,8 @@ namespace ncorr {
 				double tr = d_exx + d_eyy;
 				double eigenfirst = tr / 2 + std::sqrt(std::pow(d_eyy-d_exx, 2.0) / 4 + std::pow(d_exy, 2.0));
 				double eigensecond = tr / 2 - std::sqrt(std::pow(tr, 2.0) / 4 + std::pow(d_exy, 2.0));
-				*i_e1 = std::max(eigenfirst, eigensecond);
-				*i_e2 = std::min(eigenfirst, eigensecond);
+				*i_e1 = std::abs(eigenfirst)>std::abs(eigensecond)? eigenfirst: eigensecond;
```
Comment (markhalonen) [inline include/Strain2D.h:108]: Was this a bug?

File: src/ncorr_commandLine.cpp [hunk 2]
```diff
@@ -94,21 +83,10 @@ void exportStrainData(string outputFilesDirectory, string csvExport, strain_anal
 			}
 
 		}
-		if (exportVaribles[idx].compare("mean eyy") == 0) {
+		if (csvExport[idx].compare("mean eyy") == 0) {
```
Comment (markhalonen) [inline src/ncorr_commandLine.cpp:86]: I would prefer to use an `else if` here. It doesn't change the behavior, I just think it makes it easier to read and understand.

File: src/ncorr_commandLine.cpp [hunk 3]
```diff
@@ -526,10 +402,8 @@ int main(int argc, char *argv[]) {
 		interpType = INTERP::NEAREST;
 	}
 
-	numThreads = atof(
-		split(split(firstLevelSplit[4], "\n", false)[3], "\t")[1].c_str());
-	string subRegionstr;
-	subRegionstr = split(split(firstLevelSplit[4], "\n", false)[4], "\t")[1];
```
Comment (markhalonen) [inline src/ncorr_commandLine.cpp:532]: Definitely good to get rid of this code

File: src/ncorr_commandLine.cpp [hunk 4]
```diff
@@ -553,92 +424,105 @@ int main(int argc, char *argv[]) {
 	else{
 		config_DIC_analysis = DIC_analysis_config::REMOVE_BAD_POINTS;
 	}
+	std::vector<Image2D> imgs;
+	for (json::iterator it = inputFile.find("images")->begin(); it != inputFile.find("images")->end(); ++it) {
+  		image_names.push_back(it->get<string>());
+	}
+	//string directoryInUse = getDirPath(getImagesArray[1]);
```
Comment (markhalonen) [inline src/ncorr_commandLine.cpp]: Delete

File: src/ncorr_commandLine.cpp [hunk 5]
```diff
@@ -553,92 +424,105 @@ int main(int argc, char *argv[]) {
...
-	dispMin = atof(
-		split(split(firstLevelSplit[5], "\n", false)[18], "\t")[1].c_str());
+int main(int argc, char *argv[]) {
+	
+	if (argc != 3) {
+		throw std::invalid_argument(
+			"Must have 2 command line input of either 'calculate' or 'load' and the path of the job file");
+	}
+	try{
+	// Initialize DIC and strain information ---------------//
```
Comment (markhalonen) [inline src/ncorr_commandLine.cpp]: Indent on try-catch block missing. Any linters being used?

File: src/ncorr_commandLine.cpp [hunk 6]
```diff
@@ -797,13 +687,32 @@ int main(int argc, char *argv[]) {
 	}
 
 	else {
-		if(isEulerian)
-			saveImages(videoExport, strain_input_eulerian, getImagesArray, videoPath, strainType, DIC_output_eulerian, strain_output_eulerian, strainMax);
+		if(eulerian_video)
```
Comment (markhalonen) [inline src/ncorr_commandLine.cpp]: Prefer to have {} on this `if` statement

File: src/ncorr_commandLine.cpp [hunk 7]
```diff
@@ -797,13 +687,32 @@ int main(int argc, char *argv[]) {
...
+						strainType, 
+						DIC_output, 
+						strain_output, 
+						strainMax);
 
 	}
+	}
+	catch(const std::exception& e)
+	{
+		std::cout<<e.what()<<std::endl;
```
Comment (markhalonen) [inline src/ncorr_commandLine.cpp]: Maybe add a text error in case the exception error is cryptic

**Review by markhalonen — COMMENTED** [review]
A bunch of files have been deleted. Is the functionality implemented somewhere else?

**Comment by markhalonen** [comment]
It sounds like this isn't yet working, so I'll hold off on merging to master. You can merge if you see fit.

File: include/Strain2D.h (hunk 1 above)
Comment (bhalonen) [reply include/Strain2D.h:108]: yeah, it only mattered if you did compression. Tension results were correct.

File: src/ncorr_commandLine.cpp (hunk 2 above)
Comment (bhalonen) [reply src/ncorr_commandLine.cpp:86]: Yeah, this is ugly garbage that will get refactored fairly hard at some point. I just said screw it, get her done tonight.

File: src/ncorr_commandLine.cpp (hunk 3 above)
Comment (bhalonen) [reply src/ncorr_commandLine.cpp:532]: Yeah, slightly unmaintainable. You can see why I decided to jump to json than figure this crap out again.

File: src/ncorr_commandLine.cpp (hunk 5 above)
Comment (bhalonen) [reply src/ncorr_commandLine.cpp]: Good point. Need to clean this up more. 4 am garbage.

File: src/ncorr_commandLine.cpp (hunk 4 above)
Comment (bhalonen) [reply src/ncorr_commandLine.cpp]: What do you want me to delete here?

**Comment by bhalonen** [comment]
It is working actually. At least on my computer. Might need to do more stringent testing tomorrow.

**Comment by bhalonen** [comment]
I may do a pre-release to REL tomorrow to have them find issues.

File: include/Strain2D.h (hunk 1 above)
Comment (markhalonen) [reply include/Strain2D.h:108]: Ok, and this was only used for tension. Makes sense

File: src/ncorr_commandLine.cpp (hunk 4 above)
Comment (markhalonen) [reply src/ncorr_commandLine.cpp]: The random commented out lines of code. Unless they are meaningful?

**Comment by bhalonen** [comment]
Ah, I'll get that.

File: src/ncorr_commandLine.cpp (hunk 6 above)
Comment (bhalonen) [reply src/ncorr_commandLine.cpp]: Yeah, I guess I find it 'cute', but consistency is probably more important.

**Pull request merged**